packages/crypto/tests/signatures.test.ts

```diff
@@ -0,0 +1,100 @@
+import fs from 'node:fs'
+import * as uint8arrays from 'uint8arrays'
+import EcdsaKeypair from '../src/p256/keypair'
+import Secp256k1Keypair from '../src/secp256k1/keypair'
+import * as p256 from '../src/p256/operations'
+import * as secp from '../src/secp256k1/operations'
+import { cborEncode } from '@atproto/common'
+import {
+  bytesToMultibase,
+  multibaseToBytes,
+  parseDidKey,
+  P256_JWT_ALG,
+  SECP256K1_JWT_ALG,
+} from '../src'
+
+describe('signatures', () => {
+  let vectors: TestVector[]
+
+  beforeAll(() => {
+    vectors = JSON.parse(
+      fs.readFileSync(`${__dirname}/signature-fixtures.json`).toString(),
+    )
+  })
+
+  it('verifies secp256k1 and P-256 test vectors', async () => {
+    for (const vector of vectors) {
+      const messageBytes = uint8arrays.fromString(
+        vector.messageBase64,
+        'base64',
+      )
+      const signatureBytes = uint8arrays.fromString(
+        vector.signatureBase64,
+        'base64',
+      )
+      const keyBytes = multibaseToBytes(vector.publicKeyMultibase)
+      const didKey = parseDidKey(vector.publicKeyDid)
+      expect(uint8arrays.equals(keyBytes, didKey.keyBytes))
+      if (vector.algorithm === P256_JWT_ALG) {
+        const verified = await p256.verify(
+          keyBytes,
+          messageBytes,
+          signatureBytes,
+        )
+        expect(verified).toEqual(true)
+      } else if (vector.algorithm === SECP256K1_JWT_ALG) {
+        const verified = await secp.verify(
+          keyBytes,
+          messageBytes,
+          signatureBytes,
+        )
+        expect(verified).toEqual(true)
```

Comment:
might be worth adding a bad signature or two in here as well

like just a signature for another curve for instance